internal/sops/awskms/keysource.go

```diff
@@ -250,12 +267,13 @@ func (key MasterKey) createSTSConfig(config *aws.Config) (*aws.Config, error) {
 	if err != nil {
 		return nil, err
 	}
-	stsRoleSessionNameRe, err := regexp.Compile(stsSessionRegex)
-	if err != nil {
-		return nil, fmt.Errorf("failed to compile STS role session name regex: %w", err)
-	}
+	stsRoleSessionNameRe := regexp.MustCompile(stsSessionRegex)
+
 	sanitizedHostname := stsRoleSessionNameRe.ReplaceAllString(hostname, "")
 	name := "sops@" + sanitizedHostname
+	if len(name) >= roleSessionNameLengthLimit {
+		name = name[:roleSessionNameLengthLimit]
```

Comment:
This would not handle multi-byte strings correctly, but I am not sure how realistic this scenario is.

Reply:
TBH, I don't think the AWS SDK would even accept a multi-byte string as a valid role name, so I don't think we need to worry about that while asserting length.

Reply:
Just noticed this matches current upstream, so lets leave it.